object/ldap_conn.go

```diff
@@ -346,31 +346,52 @@ func SyncLdapUsers(owner string, syncUsers []LdapUser, ldapId string) (existUser
 	}
 	tag := strings.Join(ou, ".")
 
-	for _, syncUser := range syncUsers {
-		existUuids, err := GetExistUuids(owner, uuids)
-		if err != nil {
-			return nil, nil, err
-		}
+	ldapUuids, err := GetExistUuids(owner, uuids)
+	if err != nil {
+		return nil, err
+	}
 
-		found := false
-		if len(existUuids) > 0 {
-			for _, existUuid := range existUuids {
-				if syncUser.Uuid == existUuid {
-					existUsers = append(existUsers, syncUser)
-					found = true
+	for _, syncUser := range syncUsers {
+		foundLdapUuid := ""
+		if len(ldapUuids) > 0 {
+			for _, ldapUuid := range ldapUuids {
+				if syncUser.Uuid == ldapUuid {
+					foundLdapUuid = ldapUuid
 				}
 			}
 		}
 
 		name, err := syncUser.buildLdapUserName()
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 
-		if !found {
+		var affected bool
+		if foundLdapUuid != "" {
+			existUser, err := GetUserByLdapUuid(owner, foundLdapUuid)
```

Comment:
@theburyat try to avoid db query in loop. 
Can we fetch all needed in one query?

Reply:
fixed